<subsystem_configuration>
    <import package="controller_common" />
    <import package="rtt_geometry_msgs" />
    <import package="rtt_barrett_hand_msgs" />
    <import package="rtt_barrett_hand_subsystem_buffers" />
    <import package="rtt_velma_core_cs_ve_body_msgs" />
    <import package="rtt_velma_core_cs_ve_body_subsystem_buffers" />
    <import package="velma_core_ve_body" />
    <import package="eigen_typekit" />
    <import package="rtt_std_msgs" />
    <import package="port_operations" />
    <import package="rtt_control_msgs" />
    <import package="velma_controller" />
    <import package="rtt_cartesian_trajectory_msgs" />
    <import package="rtt_std_msgs" />
    <import package="rtt_tf" />
    <import package="rtt_velma_ec_gen_subsystem_buffers" />
    <import package="rtt_velma_ec_gen_msgs" />
    <import package="rtt_lwr_msgs" />
    <import package="barrett_hand_controller" />
    <import package="velma_common_components" />

     <component name="CameraTrigger"       type="TriggerGenerator" running="true" />

    <component name="TorsoPanTransmision"       type="SimpleTransmisionInt" />
    <component name="TorsoPanTransmisionInv"    type="SimpleTransmisionInvInt" />
    <component name="HeadPanTransmision"        type="SimpleTransmisionInt" />
    <component name="HeadPanTransmisionInv"     type="SimpleTransmisionInvInt" />
    <component name="HeadTiltTransmision"       type="SimpleTransmisionInt" />
    <component name="HeadTiltTransmisionInv"    type="SimpleTransmisionInvInt" />

    <component name="lwr_sync_r"        type="velma_core_ve_body_types::LwrStatusSync" />
    <component name="lwr_sync_l"        type="velma_core_ve_body_types::LwrStatusSync" />

    <component name="transparent_cmd"   type="velma_core_ve_body_types::TransparentCmdComponent" />
    <component name="bypass"        type="velma_core_ve_body_types::BypassCommand" />
    <component name="safe"          type="velma_core_ve_body_types::SafeComponent" />
    <component name="safe_lwr_r"    type="velma_core_ve_body_types::SafeLWR" />
    <component name="safe_lwr_l"    type="velma_core_ve_body_types::SafeLWR" />
    <component name="safe_t"        type="velma_core_ve_body_types::SafeElmo" />
    <component name="safe_hp"       type="velma_core_ve_body_types::SafeElmo" />
    <component name="safe_ht"       type="velma_core_ve_body_types::SafeElmo" />

    <component name="driver_t"      type="velma_core_ve_body_types::ElmoDriver" />
    <component name="driver_hp"     type="velma_core_ve_body_types::ElmoDriver" />
    <component name="driver_ht"     type="velma_core_ve_body_types::ElmoDriver" />

    <component name="packet_counter" type="velma_core_ve_body_types::PacketCounterComponent" />
    <component name="hw_state"      type="velma_core_ve_body_types::HwState" />

    <component name="rHand" type="BarrettHand" />
    <component name="lHand" type="BarrettHand" />

    <component name="can_queue_tx_l" type="CanQueueTxComponent" />
    <component name="can_queue_tx_r" type="CanQueueTxComponent" />

    <component name="FtSensorLeft" type="FtSensor" />
    <component name="FtSensorRight" type="FtSensor" />

    <io_buffer alias="hi_st" name="velma_core_cs_ve_body_st" />
    <io_buffer alias="hi_cmd" name="velma_core_cs_ve_body_cmd" />

    <io_buffer alias="lo_rLwr_st" name="rLwr_st" />
    <io_buffer alias="lo_rLwr_cmd" name="rLwr_cmd" />

    <io_buffer alias="lo_lLwr_st" name="lLwr_st" />
    <io_buffer alias="lo_lLwr_cmd" name="lLwr_cmd" />

    <io_buffer alias="lo_ec_st" name="EC_Status" />
    <io_buffer alias="lo_ec_cmd" name="EC_Command" />

    <ros_stream port="master_component.hi_cmd_OUTPORT"      topic="/velma_core_ve_body/hi_cmd" />
<!--    <ros_stream port="hi_stConcate.msg_OUTPORT"             topic="/velma_core_ve_body/hi_st" />-->
    <ros_stream port="master_component.lo_rLwr_st_OUTPORT"              topic="/velma_core_ve_body/lo_rLwr_st" />
    <ros_stream port="lo_rLwr_cmdConcate.msg_OUTPORT"              topic="/velma_core_ve_body/lo_rLwr_cmd" />
    <ros_stream port="master_component.lo_lLwr_st_OUTPORT"              topic="/velma_core_ve_body/lo_lLwr_st" />
    <ros_stream port="master_component.lo_ec_st_OUTPORT"                topic="/velma_core_ve_body/lo_ec_st" />
    <ros_stream port="lo_ec_cmdConcate.msg_OUTPORT"                topic="/velma_core_ve_body/lo_ec_cmd" />
    <ros_stream port="lwr_sync_r.rob_OUTPORT"              topic="/velma_core_ve_body/lo_rLwr_rob_sync" />

    <connection from="safe.safe_iterations_OUTPORT"     to="safe_hp.cycle_counter_INPORT" />
    <connection from="safe.safe_iterations_OUTPORT"     to="safe_ht.cycle_counter_INPORT" />

    <connection from="safe_t.i_OUTPORT" to="driver_t.desired_i_INPORT" />

    <connection from="safe_hp.q_OUTPORT" to="driver_hp.desired_q_INPORT" />
    <connection from="safe_hp.dq_OUTPORT" to="driver_hp.desired_dq_INPORT" />

    <connection from="safe_ht.q_OUTPORT" to="driver_ht.desired_q_INPORT" />
    <connection from="safe_ht.dq_OUTPORT" to="driver_ht.desired_dq_INPORT" />

    <connection from="lo_ec_stSplit.FTsensorleft_Data_FxGage0_OUTPORT"          to="FtSensorLeft.FxGage0_INPORT" name="ft_l_fx"/>
    <connection from="lo_ec_stSplit.FTsensorleft_Data_FyGage1_OUTPORT"          to="FtSensorLeft.FyGage1_INPORT" name="ft_l_fy"/>
    <connection from="lo_ec_stSplit.FTsensorleft_Data_FzGage2_OUTPORT"          to="FtSensorLeft.FzGage2_INPORT" name="ft_l_fz"/>
    <connection from="lo_ec_stSplit.FTsensorleft_Data_TxGage3_OUTPORT"          to="FtSensorLeft.TxGage3_INPORT" name="ft_l_tx"/>
    <connection from="lo_ec_stSplit.FTsensorleft_Data_TyGage4_OUTPORT"          to="FtSensorLeft.TyGage4_INPORT" name="ft_l_ty"/>
    <connection from="lo_ec_stSplit.FTsensorleft_Data_TzGage5_OUTPORT"          to="FtSensorLeft.TzGage5_INPORT" name="ft_l_tz"/>
    <connection from="lo_ec_stSplit.FTsensorleft_Data_StatusCode_OUTPORT"       to="FtSensorLeft.StatusCode_INPORT" name="ft_l_s"/>
    <connection from="lo_ec_stSplit.FTsensorleft_Data_SampleCounter_OUTPORT"    to="FtSensorLeft.SampleCounter_INPORT" name="ft_l_c"/>
    <connection from="FtSensorLeft.Control1_OUTPORT" to="lo_ec_cmdConcate.FTsensorleft_Control_Control1_INPORT" name="ft_l_c1"/>
    <connection from="FtSensorLeft.Control2_OUTPORT" to="lo_ec_cmdConcate.FTsensorleft_Control_Control2_INPORT" name="ft_l_c2"/>

    <connection from="FtSensorLeft.w_OUTPORT" to="hi_stConcate.lFt_rw_INPORT" />
    <connection from="FtSensorLeft.fw_OUTPORT" to="hi_stConcate.lFt_sfw_INPORT" />
    <connection from="FtSensorLeft.ffw_OUTPORT" to="hi_stConcate.lFt_ffw_INPORT" />

    <connection from="lo_ec_stSplit.FTsensorright_Data_FxGage0_OUTPORT"          to="FtSensorRight.FxGage0_INPORT" name="ft_r_fx"/>
    <connection from="lo_ec_stSplit.FTsensorright_Data_FyGage1_OUTPORT"          to="FtSensorRight.FyGage1_INPORT" name="ft_r_fy"/>
    <connection from="lo_ec_stSplit.FTsensorright_Data_FzGage2_OUTPORT"          to="FtSensorRight.FzGage2_INPORT" name="ft_r_fz"/>
    <connection from="lo_ec_stSplit.FTsensorright_Data_TxGage3_OUTPORT"          to="FtSensorRight.TxGage3_INPORT" name="ft_r_tx"/>
    <connection from="lo_ec_stSplit.FTsensorright_Data_TyGage4_OUTPORT"          to="FtSensorRight.TyGage4_INPORT" name="ft_r_ty"/>
    <connection from="lo_ec_stSplit.FTsensorright_Data_TzGage5_OUTPORT"          to="FtSensorRight.TzGage5_INPORT" name="ft_r_tz"/>
    <connection from="lo_ec_stSplit.FTsensorright_Data_StatusCode_OUTPORT"       to="FtSensorRight.StatusCode_INPORT" name="ft_r_s"/>
    <connection from="lo_ec_stSplit.FTsensorright_Data_SampleCounter_OUTPORT"    to="FtSensorRight.SampleCounter_INPORT" name="ft_r_c"/>
    <connection from="FtSensorRight.Control1_OUTPORT" to="lo_ec_cmdConcate.FTsensorright_Control_Control1_INPORT" name="ft_r_c1"/>
    <connection from="FtSensorRight.Control2_OUTPORT" to="lo_ec_cmdConcate.FTsensorright_Control_Control2_INPORT" name="ft_r_c2"/>

    <connection from="FtSensorRight.w_OUTPORT" to="hi_stConcate.rFt_rw_INPORT" />
    <connection from="FtSensorRight.fw_OUTPORT" to="hi_stConcate.rFt_sfw_INPORT" />
    <connection from="FtSensorRight.ffw_OUTPORT" to="hi_stConcate.rFt_ffw_INPORT" />

<!--    <connection from="hi_cmdSplit._OUTPORT" to="cmd_buffer.cmd_INPORT" />-->

    <connection from="hi_cmdSplit.hpMotor_q_OUTPORT" to="HeadPanTransmisionInv.JointPosition_INPORT" />
    <connection from="hi_cmdSplit.hpMotor_dq_OUTPORT" to="HeadPanTransmisionInv.JointVelocity_INPORT" />

    <connection from="hi_cmdSplit.htMotor_q_OUTPORT" to="HeadTiltTransmisionInv.JointPosition_INPORT" />
    <connection from="hi_cmdSplit.htMotor_dq_OUTPORT" to="HeadTiltTransmisionInv.JointVelocity_INPORT" />

    <connection from="lo_ec_stSplit.HeadPan_Inputs_Positionactualvalue_OUTPORT"         to="HeadPanTransmision.MotorPosition_INPORT" name="q_hp"/>
    <connection from="lo_ec_stSplit.HeadPan_Inputs_Velocityactualvalue_OUTPORT"         to="HeadPanTransmision.MotorVelocity_INPORT" name="dq_hp"/>
    <connection from="HeadPanTransmision.JointPosition_OUTPORT"         to="hi_stConcate.hpMotor_q_INPORT" />
    <connection from="HeadPanTransmision.JointVelocity_OUTPORT"         to="hi_stConcate.hpMotor_dq_INPORT" />

    <connection from="lo_ec_stSplit.HeadTilt_Inputs_Positionactualvalue_OUTPORT"        to="HeadTiltTransmision.MotorPosition_INPORT" name="q_ht"/>
    <connection from="lo_ec_stSplit.HeadTilt_Inputs_Velocityactualvalue_OUTPORT"        to="HeadTiltTransmision.MotorVelocity_INPORT" name="dq_ht"/>
    <connection from="HeadTiltTransmision.JointPosition_OUTPORT" to="hi_stConcate.htMotor_q_INPORT" />
    <connection from="HeadTiltTransmision.JointVelocity_OUTPORT" to="hi_stConcate.htMotor_dq_INPORT" />

    <connection from="lo_ec_stSplit.TorsoPan_Inputs_Positionactualvalue_OUTPORT" to="TorsoPanTransmision.MotorPosition_INPORT" name="q_t"/>
    <connection from="lo_ec_stSplit.TorsoPan_Inputs_Velocityactualvalue_OUTPORT" to="TorsoPanTransmision.MotorVelocity_INPORT" name="dq_t"/>

    <connection from="lo_rLwr_stSplit.q_OUTPORT"            to="lwr_sync_r.q_INPORT" name="q" />
    <connection from="lo_rLwr_stSplit.dq_OUTPORT"           to="lwr_sync_r.dq_INPORT" name="dq" />
    <connection from="lo_rLwr_stSplit.t_OUTPORT"            to="lwr_sync_r.t_INPORT" name="t" />
    <connection from="lo_rLwr_stSplit.gt_OUTPORT"           to="lwr_sync_r.gt_INPORT" name="gt" />
    <connection from="lo_rLwr_stSplit.w_OUTPORT"            to="lwr_sync_r.w_INPORT" name="wr" />
    <connection from="lo_rLwr_stSplit.mmx_OUTPORT"          to="lwr_sync_r.mmx_INPORT" name="M" />
    <connection from="lo_rLwr_stSplit.iState_OUTPORT"       to="lwr_sync_r.fri_INPORT" />
    <connection from="lo_rLwr_stSplit.rState_OUTPORT"       to="lwr_sync_r.rob_INPORT" />

    <connection from="lo_lLwr_stSplit.q_OUTPORT"            to="lwr_sync_l.q_INPORT" name="q" />
    <connection from="lo_lLwr_stSplit.dq_OUTPORT"           to="lwr_sync_l.dq_INPORT" name="dq" />
    <connection from="lo_lLwr_stSplit.t_OUTPORT"            to="lwr_sync_l.t_INPORT" name="t" />
    <connection from="lo_lLwr_stSplit.gt_OUTPORT"           to="lwr_sync_l.gt_INPORT" name="gt" />
    <connection from="lo_lLwr_stSplit.w_OUTPORT"            to="lwr_sync_l.w_INPORT" name="wr" />
    <connection from="lo_lLwr_stSplit.mmx_OUTPORT"          to="lwr_sync_l.mmx_INPORT" name="M" />
    <connection from="lo_lLwr_stSplit.iState_OUTPORT"       to="lwr_sync_l.fri_INPORT" />
    <connection from="lo_lLwr_stSplit.rState_OUTPORT"       to="lwr_sync_l.rob_INPORT" />

    <connection from="safe.sc_OUTPORT"              to="hi_stConcate.sc_INPORT" />
    <connection from="safe_lwr_r.t_OUTPORT"         to="lo_rLwr_cmdConcate.t_INPORT" />
    <connection from="safe_lwr_r.fri_OUTPORT"       to="lo_rLwr_cmdConcate.krlCmd_INPORT" />
    <connection from="safe_lwr_l.t_OUTPORT"         to="lo_lLwr_cmdConcate.t_INPORT" />
    <connection from="safe_lwr_l.fri_OUTPORT"       to="lo_lLwr_cmdConcate.krlCmd_INPORT" />
    <connection from="driver_t.i_OUTPORT"                 to="lo_ec_cmdConcate.TorsoPan_Outputs_TargetTorque_INPORT" />
    <connection from="driver_t.controlWord_OUTPORT"       to="lo_ec_cmdConcate.TorsoPan_Outputs_Controlword_INPORT" />
    <connection from="driver_t.modeOfOperation_OUTPORT"   to="lo_ec_cmdConcate.TorsoPan_Outputs1_Modeofoperation_INPORT" />
    <connection from="driver_hp.q_OUTPORT"                to="lo_ec_cmdConcate.HeadPan_Outputs_TargetPosition_INPORT" />
    <connection from="driver_hp.dq_OUTPORT"                to="lo_ec_cmdConcate.HeadPan_Outputs_VelocityOffset_INPORT" />
    <connection from="driver_hp.controlWord_OUTPORT"      to="lo_ec_cmdConcate.HeadPan_Outputs_Controlword_INPORT" />
    <connection from="driver_hp.modeOfOperation_OUTPORT"  to="lo_ec_cmdConcate.HeadPan_Outputs1_Modeofoperation_INPORT" />
    <connection from="driver_hp.homing_required_OUTPORT"        to="hi_stConcate.hpMotor_homing_required_INPORT" />
    <connection from="driver_hp.homing_in_progress_OUTPORT"     to="hi_stConcate.hpMotor_homing_in_progress_INPORT" />
    <connection from="driver_hp.enabled_OUTPORT"                to="hi_stConcate.hpMotor_enabled_INPORT" />
    <connection from="hi_cmdSplit.hpMotor_homing_start_OUTPORT" to="driver_hp.homing_start_INPORT" />
    <connection from="hi_cmdSplit.hpMotor_enable_OUTPORT"       to="driver_hp.enable_INPORT" />
    <connection from="safe_hp.disable_OUTPORT"                  to="driver_hp.disable_INPORT" />

    <connection from="CameraTrigger.Trigger_OUTPORT"      to="lo_ec_cmdConcate.HeadPan_Outputs_DigitalOutputs_INPORT" />
    <ros_stream port="CameraTrigger.TriggerStamp_OUTPORT" topic="/velma_core_ve_body/TriggerStamp" />

    <connection from="driver_ht.q_OUTPORT"                to="lo_ec_cmdConcate.HeadTilt_Outputs_TargetPosition_INPORT" />
    <connection from="driver_ht.dq_OUTPORT"                to="lo_ec_cmdConcate.HeadTilt_Outputs_VelocityOffset_INPORT" />
    <connection from="driver_ht.controlWord_OUTPORT"      to="lo_ec_cmdConcate.HeadTilt_Outputs_Controlword_INPORT" />
    <connection from="driver_ht.modeOfOperation_OUTPORT"  to="lo_ec_cmdConcate.HeadTilt_Outputs1_Modeofoperation_INPORT" />
    <connection from="driver_ht.homing_required_OUTPORT"        to="hi_stConcate.htMotor_homing_required_INPORT" />
    <connection from="driver_ht.homing_in_progress_OUTPORT"     to="hi_stConcate.htMotor_homing_in_progress_INPORT" />
    <connection from="driver_ht.enabled_OUTPORT"                to="hi_stConcate.htMotor_enabled_INPORT" />
    <connection from="hi_cmdSplit.htMotor_homing_start_OUTPORT" to="driver_ht.homing_start_INPORT" />
    <connection from="hi_cmdSplit.htMotor_enable_OUTPORT"       to="driver_ht.enable_INPORT" />
    <connection from="safe_ht.disable_OUTPORT"                  to="driver_ht.disable_INPORT" />

    <connection from="driver_t.homing_required_OUTPORT"        to="hi_stConcate.tMotor_homing_required_INPORT" />
    <connection from="driver_t.homing_in_progress_OUTPORT"     to="hi_stConcate.tMotor_homing_in_progress_INPORT" />
    <connection from="driver_t.enabled_OUTPORT"                to="hi_stConcate.tMotor_enabled_INPORT" />
    <connection from="hi_cmdSplit.tMotor_enable_OUTPORT"       to="driver_t.enable_INPORT" />
    <connection from="safe_t.disable_OUTPORT"                  to="driver_t.disable_INPORT" />

    <connection from="lwr_sync_r.dq_OUTPORT"    to="safe_lwr_r.dq_INPORT" name="dq" />
    <connection from="lwr_sync_r.fri_OUTPORT"   to="safe_lwr_r.friIntf_INPORT" />
    <connection from="lwr_sync_r.rob_OUTPORT"   to="safe_lwr_r.friRobot_INPORT" />
    <connection from="lwr_sync_l.dq_OUTPORT"    to="safe_lwr_l.dq_INPORT" name="dq" />
    <connection from="lwr_sync_l.fri_OUTPORT"   to="safe_lwr_l.friIntf_INPORT" />
    <connection from="lwr_sync_l.rob_OUTPORT"   to="safe_lwr_l.friRobot_INPORT" />
    <connection from="lo_ec_stSplit.TorsoPan_Inputs_Velocityactualvalue_OUTPORT"    to="safe_t.dq_INPORT" name="dq_t" />
    <connection from="lo_ec_stSplit.HeadPan_Inputs_Positionactualvalue_OUTPORT"     to="driver_hp.q_INPORT" name="q_hp" />
    <connection from="lo_ec_stSplit.HeadPan_Inputs_Positionactualvalue_OUTPORT"     to="safe_hp.q_INPORT" name="q_hp" />
    <connection from="lo_ec_stSplit.HeadTilt_Inputs_Positionactualvalue_OUTPORT"    to="driver_ht.q_INPORT" name="q_ht" />
    <connection from="lo_ec_stSplit.HeadTilt_Inputs_Positionactualvalue_OUTPORT"    to="safe_ht.q_INPORT" name="q_ht" />

    <connection from="lo_ec_stSplit.TorsoPan_Inputs_Statusword_OUTPORT"     to="driver_t.statusWord_INPORT"  name="st_t" />
    <connection from="lo_ec_stSplit.HeadPan_Inputs_Statusword_OUTPORT"      to="driver_hp.statusWord_INPORT" name="st_hp" />
    <connection from="lo_ec_stSplit.HeadTilt_Inputs_Statusword_OUTPORT"     to="driver_ht.statusWord_INPORT" name="st_ht" />

    <connection from="bypass.rArm_t_OUTPORT"        to="lo_rLwr_cmdConcate.t_INPORT" />
    <connection from="bypass.lArm_t_OUTPORT"        to="lo_lLwr_cmdConcate.t_INPORT" />

    <connection from="bypass.rHand_OUTPORT"         to="rHand.cmd_INPORT" />
    <connection from="bypass.rHandReset_OUTPORT"    to="rHand.reset_INPORT" />
    <connection from="lo_ec_stSplit.CANright_Box11CANInterfaceIN_RxQueue_OUTPORT" to="rHand.rx_queue_INPORT" name="can_r_rx"/>
    <connection from="rHand.tx_OUTPORT"             to="can_queue_tx_r.tx_INPORT" >
        <conn_policy type="circular_buffer" size="10" />
    </connection>
    <connection from="rHand.q_OUTPORT"              to="hi_stConcate.rHand_q_INPORT" />
    <connection from="rHand.status_OUTPORT"         to="hi_stConcate.rHand_s_INPORT" />

    <connection from="bypass.lHand_OUTPORT"         to="lHand.cmd_INPORT" />
    <connection from="bypass.lHandReset_OUTPORT"    to="lHand.reset_INPORT" />
    <connection from="lo_ec_stSplit.CANleft_Box10CANInterfaceIN_RxQueue_OUTPORT" to="lHand.rx_queue_INPORT" name="can_l_rx"/>
    <connection from="lHand.tx_OUTPORT"             to="can_queue_tx_l.tx_INPORT" >
        <conn_policy type="circular_buffer" size="10" />
    </connection>
    <connection from="lHand.q_OUTPORT"              to="hi_stConcate.lHand_q_INPORT" />
    <connection from="lHand.status_OUTPORT"         to="hi_stConcate.lHand_s_INPORT" />

    <connection from="lo_ec_stSplit.CANleft_Box10CANInterfaceIN_RxQueue_OUTPORT" to="can_queue_tx_l.rx_queue_INPORT" name="can_l_rx"/>
    <connection from="lo_ec_stSplit.CANright_Box11CANInterfaceIN_RxQueue_OUTPORT" to="can_queue_tx_r.rx_queue_INPORT" name="can_r_rx"/>

    <connection from="can_queue_tx_l.tx_queue_OUTPORT" to="lo_ec_cmdConcate.CANleft_Box10CANInterfaceOUT_TxQueue_INPORT" name="can_l_tx"/>
    <connection from="can_queue_tx_r.tx_queue_OUTPORT" to="lo_ec_cmdConcate.CANright_Box11CANInterfaceOUT_TxQueue_INPORT" name="can_r_tx"/>

    <connection from="transparent_cmd.sc_OUTPORT"            to="hi_stConcate.sc_INPORT" />

    <connection from="hi_cmdSplit.lArm_t_OUTPORT"                   to="bypass.lArm_t_INPORT" />
    <connection from="hi_cmdSplit.rArm_t_OUTPORT"                   to="bypass.rArm_t_INPORT" />
    <connection from="hi_cmdSplit.rHand_OUTPORT"                    to="bypass.rHand_INPORT" />
    <connection from="hi_cmdSplit.rHandReset_OUTPORT"               to="bypass.rHandReset_INPORT" />
    <connection from="hi_cmdSplit.lHand_OUTPORT"                    to="bypass.lHand_INPORT" />
    <connection from="hi_cmdSplit.lHandReset_OUTPORT"               to="bypass.lHandReset_INPORT" />

<!--
    <connection from="HeadTiltTransmisionInv.MotorPosition_OUTPORT" to="bypass.htMotor_q_INPORT" />
    <connection from="bypass.htMotor_q_OUTPORT"     to="lo_ec_cmdConcate.HeadTilt_Outputs_TargetPosition_INPORT" />
    <connection from="HeadPanTransmisionInv.MotorPosition_OUTPORT"  to="bypass.hpMotor_q_INPORT" />
    <connection from="bypass.hpMotor_q_OUTPORT"     to="lo_ec_cmdConcate.HeadPan_Outputs_TargetPosition_INPORT" />
    <connection from="TorsoPanTransmisionInv.MotorCurrent_OUTPORT"  to="bypass.tMotor_t_INPORT" />
    <connection from="bypass.tMotor_t_OUTPORT"      to="lo_ec_cmdConcate.TorsoPan_Outputs_TargetTorque_INPORT" />
-->
    <connection from="HeadTiltTransmisionInv.MotorPosition_OUTPORT"     to="driver_ht.desired_q_INPORT" />
    <connection from="HeadTiltTransmisionInv.MotorVelocity_OUTPORT"     to="driver_ht.desired_dq_INPORT" />
    <connection from="HeadPanTransmisionInv.MotorPosition_OUTPORT"      to="driver_hp.desired_q_INPORT" />
    <connection from="HeadPanTransmisionInv.MotorVelocity_OUTPORT"      to="driver_hp.desired_dq_INPORT" />
    <connection from="TorsoPanTransmisionInv.MotorCurrent_OUTPORT"      to="driver_t.desired_i_INPORT" />

    <connection from="hi_cmdSplit.tMotor_t_OUTPORT" to="TorsoPanTransmisionInv.JointTorque_INPORT" />

    <connection from="packet_counter.packet_counter_OUTPORT"            to="hi_stConcate.test_INPORT" />

    <connection from="lwr_sync_r.q_OUTPORT"            to="hi_stConcate.rArm_q_INPORT" name="q" />
    <connection from="lwr_sync_r.dq_OUTPORT"           to="hi_stConcate.rArm_dq_INPORT" name="dq" />
    <connection from="lwr_sync_r.t_OUTPORT"            to="hi_stConcate.rArm_t_INPORT" name="t" />
    <connection from="lwr_sync_r.gt_OUTPORT"           to="hi_stConcate.rArm_gt_INPORT" name="gt" />
    <connection from="lwr_sync_r.w_OUTPORT"            to="hi_stConcate.rArm_w_INPORT" name="wr" />
    <connection from="lwr_sync_r.mmx_OUTPORT"          to="hi_stConcate.rArm_mmx_INPORT" name="M" />

    <connection from="lwr_sync_l.q_OUTPORT"            to="hi_stConcate.lArm_q_INPORT" name="q" />
    <connection from="lwr_sync_l.dq_OUTPORT"           to="hi_stConcate.lArm_dq_INPORT" name="dq" />
    <connection from="lwr_sync_l.t_OUTPORT"            to="hi_stConcate.lArm_t_INPORT" name="t" />
    <connection from="lwr_sync_l.gt_OUTPORT"           to="hi_stConcate.lArm_gt_INPORT" name="gt" />
    <connection from="lwr_sync_l.w_OUTPORT"            to="hi_stConcate.lArm_w_INPORT" name="wr" />
    <connection from="lwr_sync_l.mmx_OUTPORT"          to="hi_stConcate.lArm_mmx_INPORT" name="M" />

    <connection from="TorsoPanTransmision.JointPosition_OUTPORT"        to="hi_stConcate.tMotor_q_INPORT" name="q" />
    <connection from="TorsoPanTransmision.JointVelocity_OUTPORT"        to="hi_stConcate.tMotor_dq_INPORT" name="dq" />

    <connection from="lwr_sync_r.fri_OUTPORT"       to="hw_state.rArm_fri_state_INPORT" />
    <connection from="lwr_sync_r.rob_OUTPORT"       to="hw_state.rArm_robot_state_INPORT" />
    <connection from="lwr_sync_l.fri_OUTPORT"       to="hw_state.lArm_fri_state_INPORT" />
    <connection from="lwr_sync_l.rob_OUTPORT"       to="hw_state.lArm_robot_state_INPORT" />
</subsystem_configuration>
